hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/SchemaEvolutionContext.java

```diff
@@ -139,7 +186,10 @@ public void doEvolutionForRealtimeInputFormat(AbstractRealtimeRecordReader realt
       return;
     }
     if (internalSchemaOption.isPresent()) {
-      Schema tableAvroSchema = new TableSchemaResolver(metaClient).getTableAvroSchema();
+      Schema tableAvroSchema = getAvroSchemaFromCache();
+      if (tableAvroSchema == null) {

```

Comment:
When `internalSchemaOption.isPresent()`, the `getAvroSchemaFromCache()` should also exists, this is decided by the cache.

Reply:
Yes, it seems a bit redundant, I will remove it.